jlc_kicad_tools/jlc_lib/generate_bom.py

```diff
@@ -48,62 +48,27 @@ def GenerateBOM(input_filename, output_filename, opts):
     num_groups_found = 0
     for group in grouped:
         refs = []
-        lcsc_part_numbers = set()
-        lcsc_part_numbers_none_found = False
+        lcsc_part_number = None
         footprints = set()
 
         for component in group:
             refs.append(component.getRef())
             c = component
-            lcsc_part_number = None
-
-            # Get the field name for the LCSC part number.
-            for field_name in c.getFieldNames():
-                field_value = c.getField(field_name).strip()
-
-                if LCSC_PART_NUMBER_MATCHER.match(field_value):
-                    lcsc_part_number = field_value
-
-            if lcsc_part_number:
-                lcsc_part_numbers.add(lcsc_part_number)
-            else:
-                lcsc_part_numbers_none_found = True
+            # All components in a group should have the same part number
+            lcsc_part_number = c.getLcscPartNumber()
 
             if c.getFootprint() != "":
                 footprints.add(c.getFootprint())
 
-        # Check part numbers for uniqueness
-        if len(lcsc_part_numbers) == 0 and not opts.include_all_groups:
+        if lcsc_part_number == None:
```

Comment:
Can you use "lcsc_part_number is None" for this?

Reply:
Yes, done.